src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -1282,12 +1368,12 @@ private static bool IsTypeSerializable(Type type, HashSet<Type> previousCollecti
         {
             Type itemType;
 
-            if (type.GetTypeInfo().IsEnum ||
+            if (type.IsSerializable ||
+                type.GetTypeInfo().IsEnum ||
                 type.GetTypeInfo().IsDefined(Globals.TypeOfDataContractAttribute, false) ||
                 type.GetTypeInfo().IsInterface ||
                 type.IsPointer ||
                 //Special casing DateTimeOffset and DBNull as its considered a Primitive but is no longer Serializable
-                type == Globals.TypeOfDateTimeOffset ||
```

Comment:
Comment no longer matches code, you've removed DateTimeOffset from list of checks